src/sdk/data/data.service.ts

```diff
@@ -466,5 +466,102 @@ export class DataService extends Service {
 
     return result ? result : null;
   }
-}
 
+  async getTokenLists(): Promise<TokenList[]> {
+    const { apiService } = this.services;
+
+    const { result } = await apiService.query<{
+      result: TokenLists;
+    }>(
+      gql`
+        query($chainId: Int) {
+          result: tokenLists(chainId: $chainId) {
+            items {
+              name
+              endpoint
+              isDefault
+              createdAt
+              updatedAt
+            }
+          }
+        }
+      `,
+      {
+        models: {
+          result: TokenLists,
+        },
+      },
+    );
+
+    return result.items;
+  }
+
+  async getTokenListTokens(name: string = null): Promise<TokenListToken[]> {
+    const { apiService } = this.services;
+
+    const { result } = await apiService.query<{
+      result: TokenList;
+    }>(
+      gql`
+        query($chainId: Int, $name: String) {
+          result: tokenList(chainId: $chainId, name: $name) {
+            tokens {
+              address
+              name
+              symbol
+              decimals
+              logoURI
+              chainId
+            }
+          }
+        }
+      `,
+      {
+        variables: {
+          name,
+        },
+        models: {
+          result: TokenList,
+        },
+        fetchPolicy: 'cache-first',
+      },
+    );
+
+    return result ? result.tokens : null;
```

Comment:
Is it not better to return an empty array here rather than `null`? The return type here is `Promise<TokenListToken[]` which indicates that we would at least be getting an array...

Reply:
Changes done.